Zend/zend_frameless_function.c

```diff
@@ -0,0 +1,46 @@
+/*
+   +----------------------------------------------------------------------+
+   | Zend Engine                                                          |
+   +----------------------------------------------------------------------+
+   | Copyright (c) Zend Technologies Ltd. (http://www.zend.com)           |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 2.00 of the Zend license,     |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | http://www.zend.com/license/2_00.txt.                                |
+   | If you did not receive a copy of the Zend license and are unable to  |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@zend.com so we can mail you a copy immediately.              |
+   +----------------------------------------------------------------------+
+*/
+
+#include "zend_frameless_function.h"
+
+#include <stddef.h>
+
+size_t zend_frameless_function_0_count = 0;
+size_t zend_frameless_function_1_count = 0;
+size_t zend_frameless_function_2_count = 0;
+size_t zend_frameless_function_3_count = 0;
+
+size_t zend_frameless_function_0_capacity = 0;
+size_t zend_frameless_function_1_capacity = 0;
+size_t zend_frameless_function_2_capacity = 0;
+size_t zend_frameless_function_3_capacity = 0;
+
+zend_frameless_function_0 *zend_frameless_function_0_handlers = NULL;
+zend_frameless_function_1 *zend_frameless_function_1_handlers = NULL;
+zend_frameless_function_2 *zend_frameless_function_2_handlers = NULL;
+zend_frameless_function_3 *zend_frameless_function_3_handlers = NULL;
+
+zend_function **zend_frameless_function_0_functions = NULL;
+zend_function **zend_frameless_function_1_functions = NULL;
+zend_function **zend_frameless_function_2_functions = NULL;
+zend_function **zend_frameless_function_3_functions = NULL;
+
+zend_function ** const* zend_frameless_function_functions_lists[] = {
+	&zend_frameless_function_0_functions,
+	&zend_frameless_function_1_functions,
+	&zend_frameless_function_2_functions,
+	&zend_frameless_function_3_functions,
+};
```

Comment:
Why do you need to maintain 4 different lists? One list should be enough.

Reply:
The linear search when searching the offset during compilation might increase slightly. However, that probably won't become an issue unless we have many more of them. I'll combine the lists.